Base de datos: Registros de ventas en una tienda comercial

Variable dependiente : Ventas de la tienda

In [79]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from scipy import stats
%matplotlib inline 
# ruta de datos
path='datos/Stores.csv'
df = pd.read_csv(path)
df.head()
# El tamaño del motor como potencial variable predictora del precio
# sns.regplot(x="Store_Area", y="Items_Available", data=df)
# plt.ylim(0,)
df.corr()

Store ID   Store_Area  Items_Available  \
Store ID               1.000000   -0.058705        -0.055707   
Store_Area            -0.058705    1.000000         0.998891   
Items_Available       -0.055707    0.998891         1.000000   
Daily_Customer_Count  -0.010029   -0.041423        -0.040978   
Store_Sales            0.071486    0.097474         0.098849   

                      Daily_Customer_Count  Store_Sales  
Store ID                         -0.010029     0.071486  
Store_Area                       -0.041423     0.097474  
Items_Available                  -0.040978     0.098849  
Daily_Customer_Count              1.000000     0.008629  
Store_Sales                       0.008629     1.000000

In [80]:

valor_o = stats.pearsonr(df['Store_Area'], df['Store_Sales'])
print(valor_o)

(0.09747379539480504, 0.0034936646150500283)


Dado los valores p de 0.00349 < 0.05 la correlacion entre el area de la tienda y las ventas es moderadamente significativa, el coeficiente de pearson (0.097) se acerca mas al 0 que al 1 (o -1) por tanto las variables no tiene una correlacion lineal.